st2client/tests/unit/test_command_actionrun.py

```diff
@@ -151,3 +151,27 @@ def test_get_params_from_args_with_multiple_declarations(self):
             {'foo': '1', 'bar': '2'},
             {'hoge': 'A', 'fuga': 'B'}
         ])
+
+    def test_escape_colon(self):
+        runner = RunnerType()
+        runner.runner_parameters = {}
+
+        action = Action()
+        action.ref = 'test.action'
+        action.parameters = {
+            'param_array': {'type': 'array'},
+        }
+
+        subparser = mock.Mock()
+        command = ActionRunCommand(action, self, subparser, name='test')
+
+        mockarg = mock.Mock()
+        mockarg.inherit_env = False
+        mockarg.parameters = [
+            'param_array=foo\:bar,foo2:bar2',
```

Comment:
If we decide to go with this approach, some more test cases would be great (e.g. more than two items, etc.)